### DISCUSSION TITLE
Themed Icon Support (Material You)

### DISCUSSION THREAD

**Discussion by moertel** [zixpo/candybar#124] [Ideas]
Hey everyone! Idea time :)

I know at least two Devs in this community who have monochrome icon packs. I'd love to be able to support Android 13's themed icons[1] (the ones that automatically change their tint to match wallpaper/theme colours) but I actually don't know whether that's possible from within CandyBar and what would be required.

It sounds like Google Pixel and Samsung support themed icons. On Samsung, icons can be replaced with their proprietary app Theme Park. So perhaps I only need to include adaptive icons according to Google's guide[2] to make it work?

If anyone knows more, please chime in.

[1] https://9to5google.com/2022/02/10/android-13-dp1-themed-icon-material-you/
[2] https://developer.android.com/develop/ui/views/launch/icon_design_adaptive#user-theming

**Comment by Donnnno** [comment]
Hey I would love to have something like this too!

From my understanding (this was a while ago), theming is something that has to be done on the launchers side, idk if this still the case tho. 
(https://github.com/Donnnno/Arcticons/issues/37) 

Also [Kvaesito](https://github.com/MM2-0/Kvaesitso) and [TinyBit](https://github.com/TBog/TBLauncher) can override the color already

**Reply by moertel** [reply]
Google Pixel and Samsung OneUI support this since Android 13, but only the latter has a mechanism to use custom icon packs.

From the docs, it seems that having adaptive icons and adding a `monochrome` tag is all that's needed:

```
<?xml version="1.0" encoding="utf-8"?>
...
<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="@drawable/ic_launcher_background" />
    <foreground android:drawable="@drawable/ic_launcher_foreground" />
    
    // Starting with Android 13 (API level 33), you can opt-in to providing a
    // <monochrome> drawable.
    <monochrome android:drawable="@drawable/ic_launcher_monochrome" />
</adaptive-icon>
```

So if I read this correctly, it's actually not a CandyBar thing; we could do this ourselves. 🤔 Then the only challenge is finding launchers that already support this interface.

**Reply by Donnnno** [reply]
Exactly.
 
Here are two examples of the adaptive colors applied to two totally random icons packs in Kvaesito Launcher:

![image](https://user-images.githubusercontent.com/31142286/235208262-1bdae339-4c19-4679-9cc2-65929b316a18.png) 
![image](https://user-images.githubusercontent.com/31142286/235208279-1a6a88e0-1656-457a-b5de-b3f78f887c5c.png)

**Reply by moertel** [reply]
Do you know how Kvaesito is doing it? As it's a launcher, it can probably manipulate everything it displays, but that doesn't exactly help us as icon pack developers unless there's an agreed-upon interface (like offering the `monochrome` tag in the adaptive icon).

**Comment by Donnnno** [comment]
Just noticed that this was added a while ago to Arcticons, but it turns icons themable with lawnchair and other launchers that added this intent :

```
             <!-- Lawnchair Launcher -->
            <intent-filter>
                <action android:name="app.lawnchair.icons.THEMED_ICON" />
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
```
